pkg/rules/engine/engine.go

```diff
@@ -307,3 +307,57 @@ func (engine *Engine) GetSelectedEvents() []detect.SignatureEventSelector {
 	}
 	return res
 }
+
+// GetFilters returns a mediated array of all signature's filters
+func (engine *Engine) GetFilters() ([]protocol.Filter, error) {
+	allFilters := make([]protocol.Filter, 0)
+	engine.signaturesMutex.RLock()
+	for sig := range engine.signatures {
+		filters, err := sig.GetFilters()
+		if err != nil {
+			return []protocol.Filter{}, fmt.Errorf("error getting filters: %v", err)
+		}
+		if len(filters) == 0 {
+			metadata, _ := sig.GetMetadata()
+			return []protocol.Filter{}, fmt.Errorf("error getting filters: missing filter from signature %s", metadata.Name)
+		}
+		allFilters = append(allFilters, filters...)
+	}
+	res := mediateFilters(allFilters)
+	return res, nil
+}
+
+// mediateFilters performs mediation logics on a list of filters
+func mediateFilters(filters []protocol.Filter) []protocol.Filter {
+	type filterHeader struct {
+		field    string
+		operator protocol.FilterOperator
+	}
+	remove := func(s []interface{}, i int) []interface{} {
+		s[i] = s[len(s)-1]
+		return s[:len(s)-1]
+	}
+	filterGroups := make(map[filterHeader]*protocol.Filter)
+	for _, filter := range filters {
+		header := filterHeader{field: filter.Field, operator: filter.Operator}
+		if filterGroups[header] == nil {
+			filterGroups[header] = &protocol.Filter{}
```

Comment:
why not setting `Field` and `Operator` in object creation? This instead of accessing twice to the map.